### PR TITLE
feat: enable CI to build base image for arm64 and amd64

### PR DESCRIPTION
**Pull request by guillaumeloftorbital**
**Description**
- Enable ci to build dev image  
- Enable arm64 and and64 image 
- Update CI to build package for arm64 
- slip the image build in 3 step to be able to build with the 6h job limitation 

**Testing**
- Test on core 
    https://github.com/open-space-collective/open-space-toolkit-core/actions/runs/8030904722

### CODE REVIEW

File: .github/workflows/build.yml
Comment (apaletta3) [inline .github/workflows/build.yml:1]: Can you rename this file to `build-base-image.yml`

File: .github/workflows/build.yml
Comment (apaletta3) [inline .github/workflows/build.yml:3]:
```suggestion
name: Build Base Development Image
```

File: docker/development/Dockerfile
Comment (apaletta3) [inline docker/development/Dockerfile:172]:
```suggestion
## Pybind11
```

**Review by apaletta3 — APPROVED** [review]
Couple nits, but otherwise LGTM! Thanks for this

File: .github/workflows/build.yml
Comment (guillaumeloftorbital) [reply .github/workflows/build.yml:1]: done

**Review by apaletta3 — CHANGES_REQUESTED** [review]
Can you matrix the branch as well so that we only run the arm package building for both cpp and python on main instead of on every pull request? Because the arm package building lengthens the pipeline, and we just put effort into reducing its length recently

File: docker/development/Dockerfile
Comment (vishwa2710) [inline docker/development/Dockerfile:3]: Is this necessary?

**Review by vishwa2710 — COMMENTED** [review]
Just 1 Q. Also you might need to increase the timeout as building the docker image is taking > 6 hours O_O

File: docker/development/Dockerfile
Comment (guillaumeloftorbital) [reply docker/development/Dockerfile:3]: i use the same template for build all the image and is hard coded on it https://github.com/open-space-collective/open-space-toolkit/blob/main/.github/workflows/build-image.yml

File: .github/workflows/build.yml
Comment (guillaumeloftorbital) [reply .github/workflows/build.yml]: add

File: docker/development/Dockerfile
Comment (apaletta3) [reply docker/development/Dockerfile:3]: Yes it is because he the github workflow file in each repo uses the `root-user` as the "target", and also its consistent with the other dockerfiles for all the other ostk-astro repos

**Comment by Derollez** [comment]
@guillaumeloftorbital How long does it take using the cache in the CI? Also were you able to confirm that it is able to use the cache from one build to another? 🤔 Looks great man, thanks for the awesome addition

**Comment by guillaumeloftorbital** [comment]
> e branch as well so that we only run the arm package building for both cpp and python on main instead of on every pull request? Because the arm package building lengthens the pipeline, and we just put effort into reducing its length recently

my issue with disabling the build for arm64 is that it could be easy to break the build but we will only see on release

**Comment by apaletta3** [comment]
> > e branch as well so that we only run the arm package building for both cpp and python on main instead of on every pull request? Because the arm package building lengthens the pipeline, and we just put effort into reducing its length recently
> 
> my issue with disabling the build for arm64 is that it could be easy to break the build but we will only see on release

@guillaumeloftorbital No, I don't mean only on release, I mean only on main and also on release. So if it breaks on main after we merge we'll see it right away and not have to wait for a release. We already do this in ostk-astro with validation tests (kind of like integration tests) that only run on main to keep the length of the CI down.

So could you make this change please?

**Review by apaletta3 — APPROVED** [review]
LGTM

**Pull request merged**
